lib/order.rb

```diff
@@ -0,0 +1,113 @@
+require_relative "customer"
+
+class Order
+  attr_reader :id
+  attr_accessor :products, :customer, :fulfillment_status
+
+  def initialize(id, products, customer, fulfillment_status=:pending)
+    @id = id #<-- int
+    @products= products #<-- hash
+    @customer = customer #<-- object (Customer)
+    @fulfillment_status = check_fulfillment_status(fulfillment_status) #<-- symbol
+  end
+
+  # check whether a valid fulfillment status was entered
+  # if not, raise exception
+  def check_fulfillment_status(fulfillment_status)
+    case fulfillment_status
+    when :pending, :paid, :processing, :shipped, :complete
+      return fulfillment_status
```

Comment:
I like the idea of breaking this logic out into a separate method - good instincts! One adjustment I might make is storing the list of valid statuses in a constant and using `.include?`, rather than putting them into the method itself. That way if we ever need that list elsewhere it'll be easy to get access to it.